pkg/descheduler/descheduler.go

```diff
@@ -275,6 +277,10 @@ func RunDeschedulerStrategies(ctx context.Context, rs *options.DeschedulerServer
 		} else {
 			podEvictorClient = rs.Client
 		}
+		eventBroadcaster := events.NewEventBroadcasterAdapter(rs.EventClient)
+		eventBroadcaster.StartRecordingToSink(ctx.Done())
+		eventRecorder := eventBroadcaster.NewRecorder("sigs.k8s.io.descheduler")
```

Comment:
I am creating the Recorder already here and passing it as another field in the PodEvictor struct, but if we look at how KS does it, it had a Recorder Factory with some other abstractions there: https://github.com/kubernetes/kubernetes/blob/8e14585d8700d9cf7861be66943f023d11d41075/cmd/kube-scheduler/app/server.go#L285

But I guess we should go that way only when we finish setting up the descheduler framework here

Reply:
Hmm, looking at the failing e2e tests, maybe I need the factory

Reply:
@damemi @ingvagabund @a7i I tried a few things on my local to get e2e passing, but I am not sure why e2e tests are failing here :thinking:

I initially thought I was shutting broadcaster down prematurely in e2e tests. But it seems it wasnt it. Also tried having new recorders every-time, also did not help.

Any ideas here?

Reply:
:heavy_check_mark: